src/services/apiClientFactory.ts

```diff
@@ -2,16 +2,28 @@
  * This service builds API clients.
  */
 
+import nodeFetch from "node-fetch";
+
 import { APIClient } from "../clients/api";
 import { IApiClientFactoryInterface } from "./IApiClientFactory";
 
+// TODO: this class is actually useless as APIClient is immutable, it can be removed
```

Comment:
you can remove the class safely

Reply:
the class gets used in many places, it's better to do that in a separate PR